sceptre/plan/actions.py

```diff
@@ -1144,6 +1146,19 @@ def _describe_stack_resource_drifts(self) -> dict:
             }
         )
 
+    def _filter_drifts(self, response: dict, show_only_drifted: bool) -> dict:
+        """
+        The filtered response after filtering out StackResourceDriftStatus.
+        :param show_only_drifted: Filter out IN_SYNC resources.
+        """
+        if "StackResourceDrifts" in response:
+            return {"StackResourceDrifts": [
+                drift for drift in response["StackResourceDrifts"]
+                if not show_only_drifted or drift["StackResourceDriftStatus"] != "IN_SYNC"
```

Comment:
I think this condition is overly complicated. If you just only filtered when `drifted` is true, you could simplify this. Then you can remove the boolean flag and make this function have less complexity.